deepspeed/runtime/pipe/engine.py

```diff
@@ -1207,16 +1207,19 @@ def _allocate_buffers(self, shapes_and_dtypes, requires_grad=False, num_buffers=
         return buffers
 
     def forward(self, *args, **kwargs):
-        """Disabled for pipeline parallel training. See ``train_batch()``. """
-        raise PipelineError("Only train_batch() is accessible in pipeline mode.")
+        """Disabled for pipeline parallel training. See ``train_batch()`` and ``eval_batch()``."""
```

Comment:
2. Improve error message of pipeline engine. see https://github.com/microsoft/DeepSpeed/pull/1438